test/index.js

```diff
@@ -172,15 +172,15 @@ test('TypeClass', function() {
   //  Bar :: TypeClass
   var Bar = Z.TypeClass('my-package/Bar', [Foo], hasMethod('bar'));
 
-  eq(Foo['@@type'], 'sanctuary-type-classes/TypeClass');
+  eq(Foo.constructor['@@type'], 'sanctuary-type-classes/TypeClass');
```

Comment:
I think `Z.type` could be used here, which is already tested and we could avoid direct access to ['@@type'] here.
```js
eq(Z.type(Foo), 'sanctuary-type-classes/TypeClass');
```